holidays/countries/nepal.py

```diff
@@ -0,0 +1,441 @@
+
+        # Christmas Day.
+        self._add_christmas_day("Christmas Day")
+
+        # Hindu holidays.
+
+        # Maghe Sankranti.
+        self._add_makar_sankranti("Maghe Sankranti")
+
+        # Sonam Lhochhar.
+        self._add_sonam_losar("Sonam Lhochhar")
+
+        # Maha Shivaratri.
+        self._add_maha_shivaratri("Maha Shivaratri")
+
+        # Gyalpo Lhosar.
+        self._add_gyalpo_losar("Gyalpo Lhosar")
+
+        # Fagu Poornima.
+        self._add_nepal_holi("Fagu Poornima")
+
+        # Fagu Poornima (Terai).
+        self._add_holi("Fagu Poornima (Terai)")
+
+        # Buddha Jayanti.
+        self._add_buddha_purnima("Buddha Jayanti")
+
+        # Fulpati.
+        self._add_maha_saptami("Fulpati")
+
+        # Maha Ashtami.
+        self._add_maha_ashtami("Maha Ashtami")
+
+        # Maha Navami.
+        self._add_maha_navami("Maha Navami")
+
+        # Bijaya Dashami.
+        self._add_dussehra("Bijaya Dashami")
+
+        # Ekadashi (Dashain).
+        self._add_papankusha_ekadashi("Ekadashi (Dashain)")
+
+        # Laxmi Pooja.
+        self._add_diwali_india("Laxmi Pooja")
+
+        # Gai Tihar.
+        self._add_gau_krida("Gai Tihar")
+
+        # Gobardhan Pooja.
+        self._add_govardhan_puja("Gobardhan Pooja")
+
+        # Mha Pooja.
+        self._add_govardhan_puja("Mha Pooja")
+
+        # Bhai Tika.
+        self._add_bhai_dooj("Bhai Tika")
+
+        # Chhath Parva.
+        self._add_chhath_puja("Chhath Parva")
+
+        # Tamu Lhochhar.
+        self._add_tamu_losar("Tamu Lhochhar")
+
+        # Islamic holidays.
+
+        # Eid al-Fitr.
+        self._add_eid_al_fitr_day("Id-ul-Fitr")
+
+        # Eid al-Adha.
+        self._add_eid_al_adha_day("Bakrid")
+
+        # Removed by MoHA between 2019-2020.
+        self._add_non_continuous_holidays()
+
+    def _populate_workday_holidays(self):
+        self._add_non_continuous_holidays(is_workday=True)
+
+
+class NepalIslamicHolidays(_CustomIslamicHolidays):
+    # Bakrid / Eid-al-Adha.
+    EID_AL_ADHA_DATES = {
+        2010: (NOV, 17),
+        2011: (NOV, 7),
+        2012: (OCT, 27),
+        2013: (OCT, 16),
+        2014: (OCT, 6),
+        2015: (SEP, 25),
+        2016: (SEP, 13),
+        2017: (SEP, 2),
+        2018: (AUG, 22),
+        2019: (AUG, 12),
+        2020: (AUG, 1),
+        2021: (JUL, 21),
+        2022: (JUL, 10),
+        2023: (JUN, 29),
+        2024: (JUN, 17),
+        2025: (JUN, 7),
+    }
+
+    # Id-ul-Fitr / Eid-al-Fitr.
+    EID_AL_FITR_DATES = {
+        2010: (SEP, 10),
+        2011: (AUG, 31),
+        2012: (AUG, 20),
+        2013: (AUG, 8),
+        2014: (JUL, 29),
+        2015: (JUL, 18),
+        2016: (JUL, 7),
+        2017: (JUN, 26),
+        2018: (JUN, 16),
+        2019: (JUN, 5),
+        2020: (MAY, 25),
+        2021: (MAY, 14),
+        2022: (MAY, 3),
+        2023: (APR, 22),
+        2024: (APR, 11),
+        2025: (MAR, 31),
+    }
+
+
+class NepalHinduHolidays(_CustomHinduHolidays):
```

Comment:
@KJhellico @PPsyrius I noticed this new ordering approach which is kind of inconsistent w/ other cases. I'm fine with keeping it if the appearance in the calendar order was used intentionally here (sorry for not being actively involved in this PR discussions).

Reply:
Yes, I missed it. 🤷‍♂️